SPECIFICATION.md

```diff
@@ -108,10 +108,15 @@ Interactions with the API may result in an error, these may be transport errors,
 Errors:
  
 - Must be wrapped in native errors or exceptions.
-- Should have a type that classifies the error
- 	* Transport error (could not reach the API).
- 	* API Errors (error response from API).
- 	* Validation Errors (unexpected data from API).
+- Should have a type that classifies the error:
+    * Transport Errors (could not reach the API, did not receive a response from the API).
+    * Request Errors (invalid data, bad credentials, API rejected due to invalid request, 400 class errors).
```

Comment:
If we do that, "API errors" should be "Server errors" which works for me. 

Good catch as these names should influence the names in the client libraries.